### PR TITLE
[SV Hunter] slightly improved APL

### PR DESCRIPTION
**Pull request by Meatwookie**

### CODE REVIEW

File: engine/class_modules/sc_hunter.cpp [hunk 1]
```diff
@@ -6441,60 +6441,62 @@ void hunter_t::apl_surv()
...
 
-  precombat -> add_action( "explosive_trap,if=!talent.steel_trap.enabled" );
+  precombat -> add_action( "explosive_trap" );
   precombat -> add_action( "steel_trap" );
   precombat -> add_action( "dragonsfire_grenade" );
   precombat -> add_action( "harpoon" );
 
   default_list -> add_action( "auto_attack" );
 
-  add_racial_actions( default_list );
```
Comment (Zuliane) [inline engine/class_modules/sc_hunter.cpp:6451]: we want to keep this one I suppose

Comment (nuoHep) [reply engine/class_modules/sc_hunter.cpp:6451]: no, both blood fury & berserking are added a couple lines below
if we leave this here the newly added ones will basically have no effect as the "generic" ones are less strict

File: engine/class_modules/sc_hunter.cpp [hunk 2]
```diff
@@ -6441,60 +6441,62 @@ void hunter_t::apl_surv()
...
   add_item_actions( default_list );
-
-  default_list -> add_action( "potion,name=prolonged_power,if=buff.spitting_cobra.up" );
+  
+  default_list -> add_action("berserking,if=(buff.spitting_cobra.up&buff.mongoose_fury.stack>2&buff.aspect_of_the_eagle.up)|(!talent.spitting_cobra.enabled&buff.aspect_of_the_eagle.up)");
+  default_list -> add_action("blood_fury,if=(buff.spitting_cobra.up&buff.mongoose_fury.stack>2&buff.aspect_of_the_eagle.up)|(!talent.spitting_cobra.enabled&buff.aspect_of_the_eagle.up)");
+  default_list -> add_action( "potion,name=prolonged_power,if=(talent.spitting_cobra.enabled&buff.spitting_cobra.remains)|(!talent.spitting_cobra.enabled&buff.aspect_of_the_eagle.remains" );
```
Comment (nuoHep) [inline engine/class_modules/sc_hunter.cpp]: missing arcane torrent

File: engine/class_modules/sc_hunter.cpp [hunk 3]
```diff
@@ -6441,60 +6441,62 @@ void hunter_t::apl_surv()
...
 
   default_list -> add_action( "auto_attack" );
 
-  add_racial_actions( default_list );
   add_item_actions( default_list );
-
-  default_list -> add_action( "potion,name=prolonged_power,if=buff.spitting_cobra.up" );
+  
+  default_list -> add_action("berserking,if=(buff.spitting_cobra.up&buff.mongoose_fury.stack>2&buff.aspect_of_the_eagle.up)|(!talent.spitting_cobra.enabled&buff.aspect_of_the_eagle.up)");
+  default_list -> add_action("blood_fury,if=(buff.spitting_cobra.up&buff.mongoose_fury.stack>2&buff.aspect_of_the_eagle.up)|(!talent.spitting_cobra.enabled&buff.aspect_of_the_eagle.up)");
```
Comment (nuoHep) [inline engine/class_modules/sc_hunter.cpp:6454]: missing arcane torrent

File: engine/class_modules/sc_hunter.cpp (hunk 2 above)
Comment (nuoHep) [inline engine/class_modules/sc_hunter.cpp:6455]: missing a closing brace

File: engine/class_modules/sc_hunter.cpp (hunk 1 above)
Comment (Zuliane) [reply engine/class_modules/sc_hunter.cpp:6451]: then we should remove https://github.com/simulationcraft/simc/blob/legion-dev/engine/class_modules/sc_hunter.cpp#L6290-L6295 else it's pointless, either keep the racial list and edit the changes there or erase the unneeded lines and write it below

Comment (nuoHep) [reply engine/class_modules/sc_hunter.cpp:6451]: I was actually going to drop add_racial_actions() after merging this pr as every spec wants different conditions on them anyway so there is no point in the "shared" function.

**Pull request closed without merging**